clumper/decorators.py

```diff
@@ -58,3 +60,59 @@ def wrapped(clumper, *args, **kwargs):
         return method(clumper, *args, **kwargs)
 
     return wrapped
+
+
+def multifile(param_name="path"):
+    def decorator(f):
+        sig = inspect.signature(f)
+
+        if param_name not in sig.parameters:
+            raise ValueError(f"Wrapped function has no parameter '{param_name}'")
+
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            bound_arguments = sig.bind(*args, **kwargs)
+
+            bound_arguments.apply_defaults()
+
+            path = bound_arguments.arguments[param_name]
+
+            # Using this path.Glob, extract all the files if any
+            collected_clumpers = []
+            # If path is provided as string
+            if isinstance(path, str):
+                if path.startswith("https:") or path.startswith("http:"):
+                    return f(*args, **kwargs)
+
+                if "*" not in path:
+                    return f(*args, **kwargs)
+                else:
+                    path = pathlib.Path().glob(path)
```

Comment:
Isn't this handled internally in all the functions already?

Reply:
Doesn't this `multifile` only need to trigger if there is a `*` in the filename? In all other scenarios you could argue that it  should just not do anything and just let the underlying function run.

Reply:
> Doesn't this `multifile` only need to trigger if there is a `*` in the filename? In all other scenarios you could argue that it should just not do anything and just let the underlying reader method run as-is. That suggests we might only need one if statement.

This would only work under the assumption that path is a string. As mentioned in the discussion #46, we can also have glob as input so it also needs to handle this case.

Reply:
> Isn't this handled internally in all the reader methods already?

What do you mean by this?

Reply:
Ah I was assuming that the input remain a string. Nevermind my comment.